nat.go

```diff
@@ -132,78 +135,79 @@ func (odbi *ovndb) natAddImp(lr string,Type string, externalIp string,externalMa
 		Mutations: []interface{}{mutation},
 		Where:     []interface{}{condition},
 	}
+
 	operations := []libovsdb.Operation{insertOp, mutateOp}
 	return &OvnCommand{operations, odbi, make([][]map[string]interface{}, len(operations))}, nil
 }
 
-
-//Deletes  NATs  from  router. If only router is supplied, all the
-//NATs from the logical router are deleted. If type is also speci‐
-//fied, then all the NATs that match the type will be deleted from
-//the logical router. If all the fields are given, then  a  single
-//NAT  rule that matches all the fields will be deleted. When type
-//	is snat, the ip should be  logical_ip.  When  type  is  dnat  or
-//dnat_and_snat, the ip shoud be external_ip.
-func (odbi *ovndb) natDelImp(lr string,Type string,ip ...string)(*OvnCommand, error){
+// Deletes  NATs  from  router. If only router is supplied, all the
+// NATs from the logical router are deleted. If type is also speci‐
+// fied, then all the NATs that match the type will be deleted from
+// the logical router. If all the fields are given, then  a  single
+// NAT  rule that matches all the fields will be deleted. When type
+// is snat, the ip should be  logical_ip.  When  type  is  dnat  or
+// dnat_and_snat, the ip shoud be external_ip.
+func (odbi *ovndb) lrNatDelImp(lr string, ntype string, ip ...string) (*OvnCommand, error) {
 	var operations []libovsdb.Operation
-	row := make(OVNRow)
 	var lrNatUUID string
-	if Type != ""{
-		switch Type {
-		case "snat":
-			row["type"] = Type
-			if len(ip) != 0{
-				row["logical_ip"] = ip[0]
-			}
-		case "dnat":
-			row["type"] = Type
-			if len(ip) != 0{
-			row["external_ip"] = ip[0]
+
+	row := make(OVNRow)
+
+	switch ntype {
+	case "snat":
+		row["type"] = ntype
+		if len(ip) != 0 {
+			row["logical_ip"] = ip[0]
 		}
-		case "dnat_and_snat":
-			row["type"] = Type
-			if len(ip) != 0{
-				row["external_ip"] = ip[0]
-			}
-		default:
-			return nil,ErrorSchema
+	case "dnat":
+		row["type"] = ntype
+		if len(ip) != 0 {
+			row["external_ip"] = ip[0]
 		}
-		lrNatUUID = odbi.getRowUUID(tableNAT,row)
-		if len(lrNatUUID) == 0 {
-			return nil, ErrorNotFound
+	case "dnat_and_snat":
+		row["type"] = ntype
+		if len(ip) != 0 {
+			row["external_ip"] = ip[0]
 		}
+	default:
+		return nil, ErrorOption
 	}
-	var natlist []string
-	LR ,_:= odbi.LRGet(lr)
-	if len(LR) == 0 {
+
+	lrNatUUID = odbi.getRowUUID(tableNAT, row)
+	if len(lrNatUUID) == 0 {
 		return nil, ErrorNotFound
 	}
-	for _,v := range LR  {
-		natlist = v.NAT
+
+	LRs, err := odbi.LRGet(lr)
+	if err != nil {
+		return nil, err
+	}
+	natlist := make([]string, len(LRs))
```

Comment:
when delete multiple nat it will out of range , shoud make it 
``` natlist:= make([]string,len(LRs.[0].NAT)) ```

Reply:
oh, it can't delete multiple nat ...... I try will try to find what's wrong with it.

Reply:
```odbi.getRowUUID``` just get one uuid that match row...